crates/sncast/src/helpers/block_explorer.rs

```diff
@@ -36,82 +38,191 @@ pub trait LinkProvider {
 
 impl LinkProvider for ViewBlock {
     fn transaction(&self, hash: Felt) -> String {
-        format!("{VIEWBLOCK}/tx/{hash:x}")
+        format!("{VIEWBLOCK}/tx/{hash:#x}")
     }
 
     fn class(&self, hash: Felt) -> String {
-        format!("{VIEWBLOCK}/class/{hash:x}")
+        format!("{VIEWBLOCK}/class/{hash:#x}")
     }
 
     fn contract(&self, address: Felt) -> String {
-        format!("{VIEWBLOCK}/contract/{address:x}")
+        format!("{VIEWBLOCK}/contract/{address:#x}")
     }
 }
 
 pub struct OkLink;
 
 impl LinkProvider for OkLink {
     fn transaction(&self, hash: Felt) -> String {
-        format!("{OKLINK}/tx/{hash:x}")
+        format!("{OKLINK}/tx/{hash:#x}")
     }
 
     fn class(&self, hash: Felt) -> String {
-        format!("{OKLINK}/class/{hash:x}")
+        format!("{OKLINK}/class/{hash:#x}")
     }
 
     fn contract(&self, address: Felt) -> String {
-        format!("{OKLINK}/contract/{address:x}")
+        format!("{OKLINK}/contract/{address:#x}")
     }
 }
 
 pub struct NftScan;
 
 impl LinkProvider for NftScan {
     fn transaction(&self, hash: Felt) -> String {
-        format!("{NFTSCAN}/{hash:x}")
+        format!("{NFTSCAN}/{hash:#x}")
     }
 
     fn class(&self, hash: Felt) -> String {
-        format!("{NFTSCAN}/{hash:x}")
+        format!("{NFTSCAN}/{hash:#x}")
     }
 
     fn contract(&self, address: Felt) -> String {
-        format!("{NFTSCAN}/{address:x}")
+        format!("{NFTSCAN}/{address:#x}")
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        helpers::block_explorer::Service,
+        response::{explorer_link::OutputLink, structs::DeployResponse},
+        Network,
+    };
+    use regex::Regex;
+    use starknet::macros::felt;
+    use test_case::test_case;
+
+    const MAINNET_RESPONSE: DeployResponse = DeployResponse {
+        contract_address: felt!(
+            "0x03241d40a2af53a34274dd411e090ccac1ea80e0380a0303fe76d71b046cfecb"
+        ),
+        transaction_hash: felt!(
+            "0x7605291e593e0c6ad85681d09e27a601befb85033bdf1805aabf5d84617cf68"
+        ),
+    };
+
+    const SEPOLIA_RESPONSE: DeployResponse = DeployResponse {
+        contract_address: felt!(
+            "0x0716b5f1e3bd760c489272fd6530462a09578109049e26e3f4c70492676eae17"
+        ),
+        transaction_hash: felt!(
+            "0x1cde70aae10f79d2d1289c923a1eeca7b81a2a6691c32551ec540fa2cb29c33"
+        ),
+    };
+
+    async fn assert_valid_links(input: &str) {
+        let pattern = Regex::new(r"transaction: |contract: |class: ").unwrap();
+        let links = pattern.replace_all(input, "");
+        let mut links = links.split('\n');
+
+        let contract = links.next().unwrap();
+        let transaction = links.next().unwrap();
+
+        let (contract_response, transaction_response) =
+            tokio::join!(reqwest::get(contract), reqwest::get(transaction));
+
+        assert!(contract_response.is_ok());
+        assert!(transaction_response.is_ok());
+    }
```

Comment:
Not 100% like this being reliant on the order of links and a regex. But I see that rust doesn't really support lookbehind/lookahead in regex.

Reply:
Also, order of links is only cosmetic here. This function tests link format and no other logic